### ISSUE TITLE
MobyLinuxVM Failed to Start and Initialize

### ISSUE THREAD

**Issue by BeardedYeti** [docker/for-win#350]
### Expected behavior
Successful installation of Docker for Windows

### Actual behavior
MobyLinuxVM fails to start and initialize.

### Information
- Diagnostic ID
E04D2794-EC75-4EC2-8731-DCFFC2A76CA5/2016-12-21_10-23-45

- Docker for Windows 1.12.5 (9503)
- Windows 10 Pro 64Bit
- Toolbox was used prior. Everything was successfully uninstalled including Oracle Virtual Box.

- Error Message. See log below.
```
Unable to start: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 'MobyLinuxVM' failed to start.

'MobyLinuxVM' could not initialize.

'MobyLinuxVM' failed to start. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)

'MobyLinuxVM' could not initialize. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
at Start-MobyLinuxVM, <No file>: line 262
at <ScriptBlock>, <No file>: line 363
   at Docker.Backend.HyperV.RunScript(String action, Dictionary`2 parameters)
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings)
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
```

### Steps to reproduce the behavior

  1. Installed Docker for Windows using InstallDocker.msi.
  2. Prompted to turn on Hyper-V and restart.
  3. Hyper-V turned on and Windows restarts.
  4. Docker starts at Windows start.
  5. MobyLinuxVM fails to start and initialize.

Alternatively, I've also tried starting fresh each time.

  1. Turning on Hyper-V from both Windows Shell as well as Windows Features.
  2. Windows Restarts.
  3. Installed Docker for Windows using InstallDocker.msi.
  4. Windows Restarts.
  5. Docker starts at Windows start.
  6. MobyLinuxVM fails to start and initialize.

Additionally.

- Read through many like scenarios.
- Confirmed bios and task manager virtualization each time.
- Looked at local authorization groups
- Tried creating other VM's in Hyper-V Manager. This was unsuccessful which leads me to believe it is a Hyper-V issue.
- Also created an external virtual switch in Hyper-V.

### Log
```
Version: 1.12.5 (9503)
Channel: Stable
Sha1: 3e6f00c1dcd935ed4d9507b72693b9ae0359ca50
Started on: 2016/12/21 10:14:01.464
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Pro
Edition: Professional
Id: 1511
Build: 10586
BuildLabName: 10586.672.amd64fre.th2_release_sec.161024-1825
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[10:14:01.440][Tracking       ][Info   ] Crash report and usage statistics are enabled
[10:14:01.477][SegmentApi     ][Info   ] Usage statistic: appLaunched
[10:14:01.505][NamedPipeClient][Info   ] Sending Version()...
[10:14:01.527][NamedPipeClient][Info   ] Received response for Version
[10:14:01.529][SegmentApi     ][Info   ] Usage statistic: heartbeat
[10:13:29.522][BackendServer  ][Info   ] Started
[10:13:29.522][Moby           ][Info   ] Connecting...
[10:14:01.522][NamedPipeServer][Info   ] Version()
[10:14:01.524][NamedPipeServer][Info   ] Version done.
[10:14:01.637][Notifications  ][Info   ] Docker is starting...
[10:14:01.637][Updater        ][Info   ] Checking for updates on channel Stable...
[10:14:01.649][NamedPipeClient][Info   ] Sending Start(Docker.Core.Settings)...
[10:14:01.653][NamedPipeServer][Info   ] Start(Docker.Core.Settings)
[10:14:01.658][HyperVGuids    ][Info   ] Installing GUIDs...
[10:14:01.663][HyperVGuids    ][Info   ] GUIDs installed
[10:14:01.669][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[10:14:02.117][Updater        ][Info   ] Local build 9503 is as good as the remote 9503 on channel Stable
[10:14:02.412][Firewall       ][Info   ] Ports are opened
[10:14:02.415][HyperV         ][Info   ] Create
[10:14:02.421][PowerShell     ][Info   ] Run script with parameters: -Create True -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[10:14:03.690][HyperV         ][Info   ] Enabled workaround for Build 10586 VMSwitch issue
[10:14:05.140][HyperV         ][Info   ] Creating Switch: DockerNAT...
[10:14:05.993][HyperV         ][Info   ] Switch created.
[10:14:06.296][HyperV         ][Info   ] Set IP address on switch
[10:14:07.400][HyperV         ][Info   ] Creating VM MobyLinuxVM...
[10:14:07.992][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[10:14:08.099][HyperV         ][Info   ] Creating dynamic VHD: C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx
[10:14:09.144][HyperV         ][Info   ] Attach VHD C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx
[10:14:09.403][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[10:14:09.475][HyperV         ][Info   ] Attach DVD C:\Program Files\Docker\Docker\Resources\mobylinux.iso
[10:14:09.817][HyperV         ][Info   ] VM created.
[10:14:09.837][PowerShell     ][Info   ] Run script...
[10:14:09.864][DataKit        ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.db.exe --url hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2,\\.\pipe\dockerDataBase --color never
[10:14:09.870][DataKit        ][Info   ] Started
[10:14:09.876][PowerShell     ][Info   ] Run script...
[10:14:09.891][DataKit        ][Info   ] 
[10:14:09.895][DataKit        ][Info   ] +4709us    Datakit  Starting com.docker.db.exe 09081b4 ...
[10:14:09.892][VpnKit         ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.slirp.exe --ethernet hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 --port hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 --db \\.\pipe\dockerDataBase --debug
[10:14:09.902][VpnKit         ][Info   ] Started
[10:14:09.908][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/config=static
[10:14:09.911][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/address=10.0.75.2
[10:14:09.913][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/netmask=255.255.255.0
[10:14:09.916][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/gateway=10.0.75.1
[10:14:09.919][Database       ][Info   ] Writing com.docker.driver.amd64-linux/network=hybrid
[10:14:09.922][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/docker/daemon.json={"registry-mirrors":[],"insecure-registries":[]}
[10:14:09.924][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/sysctl.conf=
[10:14:09.926][Database       ][Info   ] Writing com.docker.driver.amd64-linux/native/port-forwarding=true
[10:14:09.917][VpnKit         ][Info   ] com.docker.slirp.exe: Logging to stdout (stdout:true DEBUG:false)
[10:14:09.933][VpnKit         ][Info   ] com.docker.slirp.exe: Setting handler to ignore all SIGPIPE signals
[10:14:09.936][VpnKit         ][Info   ] com.docker.slirp.exe: vpnkit version %VERSION% with hostnet version local %HOSTNET_PINNED% uwt version 0.0.3 hvsock version 0.10.0 %HVSOCK_PINNED%
[10:14:09.939][VpnKit         ][Info   ] com.docker.slirp.exe: starting port_forwarding port_control_url:hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 vsock_path:
[10:14:09.942][VpnKit         ][Info   ] com.docker.slirp.exe: connecting to 63ee7a51-c86c-4452-a4a7-3650c64cd5e2:0B95756A-9985-48AD-9470-78E060895BE7
[10:14:09.945][VpnKit         ][Debug  ] com.docker.slirp.exe: Waiting for connections on socket hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2
[10:14:09.947][VpnKit         ][Info   ] com.docker.slirp.exe: connecting to 63ee7a51-c86c-4452-a4a7-3650c64cd5e2:30D48B34-7D27-4B0B-AAAF-BBBED334DD59
[10:14:09.950][VpnKit         ][Debug  ] com.docker.slirp.exe: Waiting for connections on socket hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2
[10:14:09.953][VpnKit         ][Info   ] com.docker.slirp.exe: hosts file has bindings for localhost localhost insight.dev
[10:14:10.265][Database       ][Info   ] Successfully wrote 8 value(s)
[10:14:10.266][DataKit        ][Info   ] 
[10:14:10.268][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/exclude
[10:14:10.273][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/http
[10:14:10.276][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/https
[10:14:10.270][DataKit        ][Info   ] +380138us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:10.332][Database       ][Info   ] Successfully removed 3 value(s)
[10:14:10.333][DataKit        ][Info   ] 
[10:14:10.340][DataKit        ][Info   ] +447032us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:10.350][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/ssl/certs/ca-certificates.crt=C:\WINDOWS\TEMP\tmp8AD.tmp
[10:14:10.435][Database       ][Info   ] Successfully wrote value
[10:14:10.436][DataKit        ][Info   ] 
[10:14:10.444][DataKit        ][Info   ] +550017us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:10.443][PowerShell     ][Info   ] Run script '$(Find-NetRoute -RemoteIPAddress 8.8.8.8).InterfaceIndex[0]'...
[10:14:10.486][Database       ][Info   ] Writing com.docker.driver.amd64-linux/slirp/dns=nameserver 192.168.1.4
nameserver 192.168.1.5
search shawmutprinting.com
[10:14:10.573][DataKit        ][Info   ] 
[10:14:10.575][DataKit        ][Info   ] +687009us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:10.573][Database       ][Info   ] Successfully wrote 1 value(s)
[10:14:10.581][HyperV         ][Info   ] Start
[10:14:10.585][PowerShell     ][Info   ] Run script with parameters: -Start True...
[10:14:10.624][HyperV         ][Info   ] Starting VM MobyLinuxVM...
[10:14:11.182][PowerShell     ][Info   ] Run script 'Import-Module Dism; Get-WindowsOptionalFeature -Online | where FeatureName -like "*Hyper-V*" | where State -ne Enabled'...
[10:14:13.719][HyperV         ][Info   ] Hyper-V is running
[10:14:13.721][HyperV         ][Info   ] Create
[10:14:13.724][PowerShell     ][Info   ] Run script with parameters: -Create True -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[10:14:13.921][HyperV         ][Info   ] Using existing Switch: DockerNAT
[10:14:13.959][HyperV         ][Info   ] Using existing Switch IP address
[10:14:13.973][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[10:14:14.085][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[10:14:14.212][HyperV         ][Info   ] VM created.
[10:14:14.216][HyperV         ][Info   ] Stop
[10:14:14.219][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[10:14:14.269][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[10:14:14.274][VpnKit         ][Info   ] Killing existing com.docker.slirp with PID 5052
[10:14:14.279][DataKit        ][Info   ] Killing existing com.docker.db with PID 9460
[10:14:14.285][HyperVGuids    ][Info   ] Installing GUIDs...
[10:14:14.288][HyperVGuids    ][Info   ] GUIDs installed
[10:14:14.291][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[10:14:15.442][Firewall       ][Info   ] Ports are opened
[10:14:15.444][HyperV         ][Info   ] Create
[10:14:15.448][PowerShell     ][Info   ] Run script with parameters: -Create True -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\mobylinux.iso...
[10:14:15.537][HyperV         ][Info   ] Using existing Switch: DockerNAT
[10:14:15.559][HyperV         ][Info   ] Using existing Switch IP address
[10:14:15.572][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[10:14:15.806][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[10:14:15.902][HyperV         ][Info   ] VM created.
[10:14:15.907][PowerShell     ][Info   ] Run script...
[10:14:15.920][DataKit        ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.db.exe --url hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2,\\.\pipe\dockerDataBase --color never
[10:14:15.924][DataKit        ][Info   ] Started
[10:14:15.928][PowerShell     ][Info   ] Run script...
[10:14:15.938][DataKit        ][Info   ] 
[10:14:15.942][VpnKit         ][Info   ] Starting C:\Program Files\Docker\Docker\Resources\com.docker.slirp.exe --ethernet hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 --port hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 --db \\.\pipe\dockerDataBase --debug
[10:14:15.946][VpnKit         ][Info   ] Started
[10:14:15.942][DataKit        ][Info   ] +3618us    Datakit  Starting com.docker.db.exe 09081b4 ...
[10:14:15.950][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/config=static
[10:14:15.958][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/address=10.0.75.2
[10:14:15.961][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/netmask=255.255.255.0
[10:14:15.964][Database       ][Info   ] Writing com.docker.driver.amd64-linux/net/gateway=10.0.75.1
[10:14:15.966][Database       ][Info   ] Writing com.docker.driver.amd64-linux/network=hybrid
[10:14:15.969][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/docker/daemon.json={"registry-mirrors":[],"insecure-registries":[]}
[10:14:15.972][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/sysctl.conf=
[10:14:15.975][Database       ][Info   ] Writing com.docker.driver.amd64-linux/native/port-forwarding=true
[10:14:15.958][VpnKit         ][Info   ] com.docker.slirp.exe: Logging to stdout (stdout:true DEBUG:false)
[10:14:15.982][VpnKit         ][Info   ] com.docker.slirp.exe: Setting handler to ignore all SIGPIPE signals
[10:14:15.985][VpnKit         ][Info   ] com.docker.slirp.exe: vpnkit version %VERSION% with hostnet version local %HOSTNET_PINNED% uwt version 0.0.3 hvsock version 0.10.0 %HVSOCK_PINNED%
[10:14:15.988][VpnKit         ][Info   ] com.docker.slirp.exe: starting port_forwarding port_control_url:hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2 vsock_path:
[10:14:15.991][VpnKit         ][Info   ] com.docker.slirp.exe: connecting to 63ee7a51-c86c-4452-a4a7-3650c64cd5e2:0B95756A-9985-48AD-9470-78E060895BE7
[10:14:15.994][VpnKit         ][Debug  ] com.docker.slirp.exe: Waiting for connections on socket hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2
[10:14:15.997][VpnKit         ][Info   ] com.docker.slirp.exe: connecting to 63ee7a51-c86c-4452-a4a7-3650c64cd5e2:30D48B34-7D27-4B0B-AAAF-BBBED334DD59
[10:14:15.999][VpnKit         ][Debug  ] com.docker.slirp.exe: Waiting for connections on socket hyperv-connect://63ee7a51-c86c-4452-a4a7-3650c64cd5e2
[10:14:16.002][VpnKit         ][Info   ] com.docker.slirp.exe: hosts file has bindings for localhost localhost insight.dev
[10:14:16.201][Database       ][Info   ] Successfully wrote 8 value(s)
[10:14:16.204][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/exclude
[10:14:16.206][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/http
[10:14:16.208][Database       ][Info   ] Removing com.docker.driver.amd64-linux/proxy/https
[10:14:16.202][DataKit        ][Info   ] 
[10:14:16.215][DataKit        ][Info   ] +267478us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:16.249][DataKit        ][Info   ] 
[10:14:16.251][DataKit        ][Info   ] +314355us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:16.249][Database       ][Info   ] Successfully removed 3 value(s)
[10:14:16.267][Database       ][Info   ] Writing com.docker.driver.amd64-linux/etc/ssl/certs/ca-certificates.crt=C:\WINDOWS\TEMP\tmp1FD0.tmp
[10:14:16.351][Database       ][Info   ] Successfully wrote value
[10:14:16.354][PowerShell     ][Info   ] Run script '$(Find-NetRoute -RemoteIPAddress 8.8.8.8).InterfaceIndex[0]'...
[10:14:16.352][DataKit        ][Info   ] 
[10:14:16.360][DataKit        ][Info   ] +417529us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:16.374][Database       ][Info   ] Writing com.docker.driver.amd64-linux/slirp/dns=nameserver 192.168.1.4
nameserver 192.168.1.5
search shawmutprinting.com
[10:14:16.456][DataKit        ][Info   ] 
[10:14:16.458][DataKit        ][Info   ] +521356us       fs9p [ERROR] dispatcher caught Unix.Unix_error(Unix.EPIPE, "bytes_read", ""): no more requests will be handled
[10:14:16.456][Database       ][Info   ] Successfully wrote 1 value(s)
[10:14:16.465][HyperV         ][Info   ] Start
[10:14:16.468][PowerShell     ][Info   ] Run script with parameters: -Start True...
[10:14:16.501][HyperV         ][Info   ] Starting VM MobyLinuxVM...
[10:14:17.061][NamedPipeServer][Error  ] Unable to execute Start: Unable to start: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 'MobyLinuxVM' failed to start.

'MobyLinuxVM' could not initialize.

'MobyLinuxVM' failed to start. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)

'MobyLinuxVM' could not initialize. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
at Start-MobyLinuxVM, <No file>: line 262
at <ScriptBlock>, <No file>: line 363    at Docker.Backend.HyperV.RunScript(String action, Dictionary`2 parameters)
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings)
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass8_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
[10:14:17.095][NamedPipeClient][Error  ] Unable to send Start: Unable to start: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 'MobyLinuxVM' failed to start.

'MobyLinuxVM' could not initialize.

'MobyLinuxVM' failed to start. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)

'MobyLinuxVM' could not initialize. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
at Start-MobyLinuxVM, <No file>: line 262
at <ScriptBlock>, <No file>: line 363
[10:14:17.106][Notifications  ][Error  ] Unable to start: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 'MobyLinuxVM' failed to start.

'MobyLinuxVM' could not initialize.

'MobyLinuxVM' failed to start. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)

'MobyLinuxVM' could not initialize. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
at Start-MobyLinuxVM, <No file>: line 262
at <ScriptBlock>, <No file>: line 363
[10:14:21.150][NamedPipeClient][Info   ] Sending GetDebugInfo()...
[10:14:21.159][NamedPipeServer][Info   ] GetDebugInfo()
[10:14:23.252][ErrorReportWindow][Info   ] Open logs
```

**Comment by reypm** [comment]
This is likely the same issue as reported in https://github.com/docker/for-win/issues/347

**Comment by neridonk** [comment]
Same problem help me pls

**Comment by jpayne** [comment]
Same issue here. Windows 10, Docker stable release.

**Comment by jpayne** [comment]
I seem to have the Docker stable build working now.

Hyper-V apparently does not like the Samsung 960 Pro I just installed in my ThinkPad. Trying to create a virtual disk (either via Docker or the Hyper-V Manager) on the 960 Pro fails, sometimes with a BSOD attributed to "secnvme.sys". Fortunately, I also have a Samsung 850 Pro in this system, so I configured Hyper-V to use it by default, and virtual disk creation now succeeds. I also deleted all Docker app data and reinstalled Docker.

**Comment by rneugeba** [comment]
@reypm the error message in your logs is different. It says:
```
[20:50:52.459][Linux          ][Error  ] Failed to start: Unable to create: The running command stopped because the preference variable "ErrorActionPr
eference" or common parameter is set to Stop: The computer 'REYNIERPM-DESKT' could not be resolved. Make sure you typed the machine name correctly and
 that you have network access.
at Get-Vhd-Root, <No file>: line 93
at New-MobyLinuxVM, <No file>: line 214
at <ScriptBlock>, <No file>: line 374. Let's retry
```
I also see that it seem to have worked with the 1.12.5-stable...could you please open a separate issue with the details?

**Comment by rneugeba** [comment]
@neridonk could you provide a diagnostics ID so we can take a look?

**Comment by rneugeba** [comment]
@BeardedYeti Thanks for the detailed report. Creating an external switch should not be an issue. I have that setup as well.

The Hyper-V event log is not very helpful:
```
12/21/2016 10:14:17 AM       15130 Error            'MobyLinuxVM' failed to start. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
12/21/2016 10:14:16 AM        3040 Error            'MobyLinuxVM' could not initialize. (Virtual machine ID 63EE7A51-C86C-4452-A4A7-3650C64CD5E2)
```

If you can't create other VMs with Hyper-V manager that is, of course an issue. SO that should be looked at first...

Can you try creating an empty VM, basically create a new VM and then click through `Next` on every step without changing any options. Then, select the new VM and click on `Connect`. In the new window power on the VM. You should see something like this:
<img width="640" alt="screen shot 2016-12-28 at 19 44 59" src="https://cloud.githubusercontent.com/assets/3338098/21530143/85121544-cd36-11e6-9618-ef1bd285f16d.png">

If not could you supply any error message printed?

The error messages in the Hyper-V log above both for [Event ID 3040](https://technet.microsoft.com/en-us/library/dd581971(v=ws.10).aspx) and [Event ID 15130](https://technet.microsoft.com/en-us/library/dd582049(v=ws.10).aspx) seem to suggest either permission issues or issue with the VMM service. Your system as the VMM service (`vmms`) running, which can be an issue on some systems, but it might be worth restarting it as suggested in the links and retry.

Finally, you seem to have a Skylake processor which has some new features (we saw some other. unrelated issues with it and Hyper-V) so it's worth making sure that you have all windows updates installed.

**Comment by statdev** [comment]
rneugeba:
Im having the same issue on W10. The applet fails to start, complaining about memory and launching the VM fails due to memory. Ive got quite a bit of RAM left, so I figured Id try running another VM. 

I was able to run multiple VMs without an issue, whatever is wrong lies within Docker.

**Comment by rneugeba** [comment]
Closing this issue due to inactivity. @statdev please open a separate issue with a diagnostics ID and the latest Beta or stable release. @BeardedYeti please re-open if you still have the issue and have tried the suggestion above.
thanks

**Comment by BM78** [comment]
I have also encountered this issue with a Samsung 960 Pro.  It happens when I create a new standalone VHDX as well.  I plugged in an 850 EVO via USB, pointed the Hyper-V virtual disk path at it and was able to get things working.

**Comment by jpayne** [comment]
This failure seems to be caused by Samsung's NVMe 2.0 driver for the 960 Pro. Reverting to Microsoft's NVMe driver caused VM creation to start working again. Seems Microsoft and/or Samsung have a Hyper-V incompatibility issue to sort out.

**Comment by kjwise** [comment]
@jpayne I can confirm. Happens to me as well.
There's a thread on TechNet: https://social.technet.microsoft.com/Forums/en-US/858acc34-14f6-4b0c-b040-226b1aa83d11/hyperv-crashes-when-using-the-samsung-nmve-driver-with-the-960-pro-ssd

**Comment by okproject** [comment]
Samsung again! Never ever samsung again!

**Comment by gund** [comment]
I was using Magician to keep drivers for 960 PRO up to date, but today installed Docker on Win10PRO and got this BSOD error...
Apparently Magician was doing poor job of keeping drivers up to date - once I manually grabbed drivers v2.2 the issue was fixed...
What's wrong with Samsung this days?..

**Comment by thadG** [comment]
I had similar problems using docker on Windows10

The error was :
 ""Unable to create: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Failed to create the virtual hard disk. The system failed to create 'C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx'.""

I removed docker from Windows and also found the file 'MobyLinuxVM.vhdx' located at 'C:\Users\Public Documents\Hyper-V\Virtual hard disks\ was corrupted.  It's memory size was 6,569,023KB. I deleted MobyLinuxVM.vhdx.  After reloading a Docker, I restarted the computer.

The restart reloaded  'MobyLinuxVM.vhdx' located at 'C:\Users\Public Documents\Hyper-V\Virtual hard disks\ and now it was only 4,096 KB now. This was a good sign.

I right clicked on the  'Docker for Windows' icon and ran it as an administrator, but I still had the same error. I restarted the computer again.  After the restart, Docker started to work again.
